localization/ekf_localizer/src/ekf_localizer.cpp

```diff
@@ -446,7 +446,15 @@ void EKFLocalizer::measurementUpdatePose(const geometry_msgs::msg::PoseWithCovar
     poseMeasurementCovariance(pose.pose.covariance, params_.pose_smoothing_steps);
 
   ekf_.updateWithDelay(y, C, R, delay_step);
-  updateSimple1DFilters(pose, params_.pose_smoothing_steps);
+
+  // Considering change of z value due to measurement pose delay
+  const auto rpy = tier4_autoware_utils::getRPY(pose.pose.pose.orientation);
+  const double dz_delay = calculateDeltaZFromPitch(current_ekf_twist_, delay_time, rpy.y);
```

Comment:
IMHO you don't need to create a dedicated function for this, since the internal calculation can be written in one liner.